src/sentry/identity/vsts/provider.py

```diff
@@ -135,3 +135,108 @@ def exchange_token(self, request: Request, pipeline, code):
         if req.headers["Content-Type"].startswith("application/x-www-form-urlencoded"):
             return dict(parse_qsl(body))
         return orjson.loads(body)
+
+
+class VSTSNewIdentityProvider(OAuth2Provider):
+    key = "vsts_new"
+    name = "Azure DevOps"
+
+    oauth_access_token_url = "https://login.microsoftonline.com/common/oauth2/v2.0/token"
+    oauth_authorize_url = "https://login.microsoftonline.com/common/oauth2/v2.0/authorize"
+
+    def get_oauth_client_id(self):
+        return options.get("vsts_new.client-id")
+
+    def get_oauth_client_secret(self):
+        return options.get("vsts_new.client-secret")
+
+    def get_refresh_token_url(self):
+        return self.oauth_access_token_url
+
+    def get_pipeline_views(self):
+        return [
+            VSTSOAuth2LoginView(
```

Comment:
created a new view below

Reply:
Why didn't the old one need a specific view and this one does?

Reply:
the `VSTSOAuth2LoginView` overwrites some parts of the normal `OAuthLogin2View`, which it subclasses. similarly for the callback view